workspace/src/app/views/shared/TaskActivityOverview/TaskActivityOverview.tsx

```diff
@@ -105,6 +113,24 @@ export function TaskActivityOverview({ projectId, taskId }: IProps) {
         );
     };
 
+    const statusMapDependency = Array.from(activityStatusMap.values())
+        .map((v) => `${v.lastUpdateTime}-${v.concreteStatus}-${v.progress}`)
+        .join("|");
+
+    React.useEffect(() => {
+        const currentlyRunningActivity = Array.from(activityStatusMap.values()).find((a) => a.isRunning);
+        const activity = cacheActivities[cursor.current];
+        //if no currently running activity then call the next cursor
+        if (!currentlyRunningActivity && activity) {
+            const activityFunctions = activityFunctionsCreator(activity);
+            activityFunctions.executeActivityAction("restart");
```

Comment:
Currently this will display an error if the activity is already running. It should either restart the activity again after it has finished or ignore the error if it is OK that the previous restart suffices.